core/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorTests.java

```diff
@@ -76,6 +77,33 @@ public void testNoAsyncFetchData() {
     }
 
     /**
+     * Verifies that on index creation, we don't go and fetch data, but keep the replica shard unassigned to let
+     * the shard allocator to allocate it. There isn't a copy around to find anyhow.
+     */
+    @Test
+    public void testNoAsyncFetchOnIndexCreation() {
+        RoutingAllocation allocation = onePrimaryOnNode1And1Replica(yesAllocationDeciders(), Settings.EMPTY, UnassignedInfo.Reason.INDEX_CREATED);
+        testAllocator.clean();
+        testAllocator.allocateUnassigned(allocation);
+        assertThat(testAllocator.getFetchDataCalledAndClean(), equalTo(false));
+        assertThat(allocation.routingNodes().shardsWithState(ShardRoutingState.UNASSIGNED).size(), equalTo(1));
+        assertThat(allocation.routingNodes().shardsWithState(ShardRoutingState.UNASSIGNED).get(0).shardId(), equalTo(shardId));
+    }
+
+    /**
+     * Verifies that for anything but index creation, fetch data ends up being called, since we need to go and try
+     * and find a better copy for the shard.
+     */
+    @Test
+    public void testAsyncFetchOnAnythingButIndexCreation() {
+        UnassignedInfo.Reason reason = UnassignedInfo.Reason.values()[randomIntBetween(1, UnassignedInfo.Reason.values().length - 1)];
```

Comment:
can we do randomFrom(EnumSet.complementOf(EnumSet.of(UnassignedInfo.Reason.INDEX_CREATED))) ?